code/modules/projectiles/ammunition.dm

```diff
@@ -41,6 +41,29 @@
 
 	update_icon()
 
+/obj/item/ammo_casing/Crossed(atom/AM)
+	..()
+
+	if(isliving(AM))
+		var/mob/living/L = AM
+
+		if(L.buckled)
+			return
+
+		if(!MOVING_DELIBERATELY(L) && prob(10))
+			on_casing_cross(L)
+
+/obj/item/ammo_casing/proc/on_casing_cross(atom/AM)
```

Comment:
Why make this a separate proc? Just adds proc overhead and a redundant check on `AM`.